slimta/smtp/server.py

```diff
@@ -260,10 +260,13 @@ def _command_EHLO(self, ehlo_as):
             bad_sequence.send(self.io)
             return
 
-        ehlo_as = ehlo_as.decode('utf-8')
-        reply = Reply('250', 'Hello '+ehlo_as)
-        reply.enhanced_status_code = False
-        self._call_custom_handler('EHLO', reply, ehlo_as)
+        if ehlo_as is None:
+            reply = bad_arguments
+        else:
+            ehlo_as = ehlo_as.decode('utf-8')
+            reply = Reply('250', 'Hello '+ehlo_as)
+            reply.enhanced_status_code = False
+            self._call_custom_handler('EHLO', reply, ehlo_as)
```

Comment:
I think I'd rather see this implemented as if/elif clauses at the top of the handlers, e.g.

```python
if not self.bannered:
    bad_sequence.send(self.io)
    return
elif not ehlo_as:
    bad_arguments.send(self.io)
    return
```